### ISSUE TITLE
No call graphs if there are more than one using namespace directives.

### ISSUE THREAD

**Issue by whatacold** [doxygen/doxygen#8011]
**Describe the bug**
1. Call graphs of class methods aren't generated if there are more than one `using namespace` directives in C++.

   See a minimum demo in the attachment file.

2. Referencing macros in method signatures confuse doxygen.

   For example,

   ```
   #define EXPORT(x)
   class Test {
   public:
       EXPORT(a) void foo(); // doxygen thinks EXPORT is a method!
   };
   ```

**Expected behavior**
1. Call graphs should be generated.
2. Methods are recognized correctly.

**Screenshots**
NA

**To Reproduce**
See the attachment 
[doxygen-callgraph.zip](https://github.com/doxygen/doxygen/files/5177704/doxygen-callgraph.zip).

**Version**
doxygen 1.8.15 on Fedora 30, installed via `dnf`.

**Stack trace**
NA

**Additional context**
NA

**Comment by albert-github** [comment]
You are mentioning in your issue 2 issues, as they (and here clearly) are independent problems they should have been reported in 2 separate issues. Also the example you gave only shows the first issue.

- I think that here doxygen gets confused whether or not `bar` could also be in the `namespace std` as doxygen has no information about the contents of `std` (although for the human reader it is clear here). 
  Also, as a test, when adding in `test.h`
  ```
  namespace demo1 {
    class Test1 {
      public:
         void foo1();
         void bar1();
    };
  }                           
  ```
  and changing the (in `test.cpp`):
  ```
  using namespace std; 
  ```
  into
  ```
  using namespace demo1; 
  ```
and adding:
  ```
  void Test1::foo1()
  {
      bar1();
  }

  void Test1::bar1()
  {
      return;
  }
  ```
  we don't get the call graph, although it is here, as far as I can see, unambiguous.
  I see the problems also with doxygen 1.8.20 
- the problem with the macro `EXPORT`
   in your doxygen configuration file you have set `MACRO_EXPANSION        = NO`, resulting in the behavior you descripbe. The value should be set to `YES`.

**Comment by whatacold** [comment]
Hi Albert,

Thanks for testing it with version 1.8.20, and adding another test case.

As for the second one, setting `MACRO_EXPANSION = YES` does solve the problem, thanks.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
